.github/workflows/ci.yml

```diff
@@ -128,41 +185,44 @@ jobs:
           max_attempts: 5
           retry_wait_seconds: 30
           command: |
-            composer update --optimize-autoloader --no-interaction --no-progress ${{ matrix.composer-flags }}
+            composer update --optimize-autoloader --no-interaction --no-progress --no-scripts ${{ matrix.composer-flags }}
             composer info -D
 
       # execute migration rules before running tests and self-fixing,
       # so that we know that our codebase is future-ready
       - name: Run PHP CS Fixer with migration rules
-        if: matrix.migration-rules == 'yes'
+        if: matrix.run-migration-rules == 'yes'
         run: php php-cs-fixer fix --config .php-cs-fixer.php-highest.php -q
 
       - name: Disable time limit for tests when collecting coverage
-        if: matrix.calculate-code-coverage == 'yes'
+        if: matrix.code-coverage == 'yes'
         run: sed 's/enforceTimeLimit="true"/enforceTimeLimit="false"/g' phpunit.xml.dist > phpunit.xml
 
       - name: Run tests
-        if: matrix.calculate-code-coverage != 'yes'
+        if: matrix.run-tests == 'yes'
         env:
           PHP_CS_FIXER_IGNORE_ENV: ${{ matrix.PHP_CS_FIXER_IGNORE_ENV }}
           FAST_LINT_TEST_CASES: ${{ matrix.FAST_LINT_TEST_CASES }}
         run: vendor/bin/paraunit run ${{ matrix.phpunit-flags || '--exclude-group auto-review' }}
 
       - name: Collect code coverage
-        if: matrix.calculate-code-coverage == 'yes'
+        if: matrix.code-coverage == 'yes'
+        env:
+          FAST_LINT_TEST_CASES: 1
```

Comment:
I extracted 2 changes to separated PRs, so they can get merged asap.
This PR is ~2 weeks old and there is no need for them to wait more, while other elements of this PR are still under discussion

- #7389
- #7390